### PR TITLE
Snapshot rke2 

### PR DESCRIPTION
**Pull request by fmoral2**
#### Testing ####

Checklist:
1. If your PR changes anything on or related to Jenkins, run it pointing to your branch to make sure it's okay.

2. Verify code lint; we should not have errors.

3. Update the documentation if needed.

4. Update makefile and docker run if adding new tests.

5. Run your tests at least 4 times with all configurations needed and possible.

6. If needed test with different os types.

### CODE REVIEW

File: pkg/testcase/snapshotwebhook.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	var workloadErr error
+	if applyWorkload {
+		workloadErr = shared.ManageWorkload("apply", "snapshot-webhook.yaml")
+		Expect(workloadErr).NotTo(HaveOccurred(), "Snapshot Webhook manifest not deployed")
+	}
+
+	cmd := "kubectl get endpoints -n kube-system rke2-snapshot-validation-webhook --kubeconfig=" +
+		shared.KubeConfigFile
+
+	ipPattern := `\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}:8443`
```
Comment (ShylajaDevadiga) [inline pkg/testcase/snapshotwebhook.go:22]: Let's add two assertions 1. ep should not be empty 2. workload should return somthing like this `
Error from server: error when creating "STDIN": admission webhook "rke2-snapshot-validation-webhook.csi.kubernetes.io" denied the request: Spec.VolumeSnapshotClassName must not be the empty string`

Comment (fmoral2) [reply pkg/testcase/snapshotwebhook.go]: 1. ep should not be empty 

ip should not be empty ?  @ShylajaDevadiga

File: pkg/testcase/snapshotwebhook.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+		workloadErr := shared.ManageWorkload("apply", "snapshot-webhook.yaml")
+		if workloadErr != nil {
+			if strings.Contains(workloadErr.Error(), assert) {
+				shared.LogLevel("error", workloadErr.Error())
+				shared.LogLevel("info", "Snapshot Webhook manifest not deployed, "+
+					"as expected related to empty string")
+				return workloadErr
+			}
+			shared.LogLevel("error", workloadErr.Error(),
+				"Error not related to empty string, please double check")
```
Comment (ShylajaDevadiga) [inline pkg/testcase/snapshotwebhook.go:22]: We could log this as "Error: webhook not rejecting bad VolumeSnapshot"

**Review by ShylajaDevadiga — DISMISSED** [review]
Small change in log message. otherwise LGTM

File: pkg/testcase/snapshotwebhook.go (hunk 1 above)
Comment (ShylajaDevadiga) [reply pkg/testcase/snapshotwebhook.go]: yes. here is an example 
```
kubectl get endpoints -n kube-system rke2-snapshot-validation-webhook
NAME                               ENDPOINTS   AGE
rke2-snapshot-validation-webhook               107m
```
Expected output
```
kubectl get endpoints -n kube-system rke2-snapshot-validation-webhook
NAME                               ENDPOINTS           AGE
rke2-snapshot-validation-webhook   10.42.96.137:8443   77m
```

File: pkg/testcase/snapshotwebhook.go (hunk 2 above)
Comment (fmoral2) [reply pkg/testcase/snapshotwebhook.go]: done

**Review by ShylajaDevadiga — APPROVED** [review]

**Review by mdrahman-suse — APPROVED** [review]
LGTM

**Pull request merged**